app.py

```diff
@@ -0,0 +1,172 @@
+            result = client.conversations_history(
+                channel=channel_id,
+                inclusive=True,
+                oldest=message_ts,
+                latest=message_ts,
+                limit=1
+            )
+            message = result["messages"][0]
+
+            # Print message text
+            logger.info("{} has reacted with a '{}' to {}".format(user_id, reaction, message["text"]))
+            client.chat_postMessage(channel=channel_id, text="What do you want to title this wiki?", thread_ts=message_ts)
+
+            #  writes the message that was being reacted to into the messages.txt file
+            file.write("{}\n".format(message["text"]))
+            file.close()
+
+        except SlackApiError as e:
+            logger.info(f"Error: {e}")
+
+
+
+# =============== Pin Added Events ================ #
+# When a users pins a message the type of the event will be 'pin_added'.
+# Here we'll link the update_pin callback to the 'pin_added' event.
+@app.event("pin_added")
+def update_pin(event, client):
+    """Update the onboarding welcome message after receiving a "pin_added"
+    event from Slack. Update timestamp for welcome message as well.
+    """
+    # Get the ids of the Slack user and channel associated with the incoming event
+    channel_id = event.get("channel_id")
+    user_id = event.get("user")
+
+    # Get the original tutorial sent.
+    onboarding_tutorial = onboarding_tutorials_sent[channel_id][user_id]
+
+    # Mark the pin task as completed.
+    onboarding_tutorial.pin_task_completed = True
+
+    # Get the new message payload
+    message = onboarding_tutorial.get_message_payload()
+
+    # Post the updated message in Slack
+    updated_message = client.chat_update(**message)
+
+
+# ============== Message Events ============= #
+# When a user sends a message to a channel, the event type will be 'message'.
+# Here we'll link the message callback to the 'message' event.
+@app.event("message")
+def message(event, client):
+    # initializes the file for the messages and puts cursor at the last line
+    file = open("PythOnBoardingBot/messages.txt", "r+")
+    file.seek(len(file.read(-1)))
+
+    channel_id = event.get("channel")
+    user_id = event.get("user")
+    text = event.get("text")
+    message_ts = event.get("ts")
+    message_thread_ts = event.get("thread_ts")
+
+    # logs the message in the terminal, but NOT write it to a text file.
+    if text and text.lower() == "start":
+        return start_onboarding(user_id, channel_id, client)
+
+    logger.info("{} sent message: {}".format(user_id, text))
+
+    # if the message is sent as a threaded reply within ten minutes of the original message, 
+    # it'll be written into messages.txt as the title of documentation. Will fix later.
+    try:    
+        print(float(message_ts) - float(message_thread_ts))
+        if float(message_ts) - float(message_thread_ts) <= 600:

```

Comment:
Since each page will only have one title, and each section only one heading, have you considered storing whether you have already scraped a title from that thread instead of using time?

Reply:
Sorry for not having responded to this before I started working on changing how the bot scrapes messages. I did take the idea and altered it slightly to have the code use the timestamp of a message to check whether the message has been scraped already. This was a much bigger change than just a few lines. Do I need to update the code in this PR specifically or can I put the updated code in the next PR?

Reply:
Feel free to stick this in another PR! Just was throwing out an idea- will mark this as resolved.